### PR TITLE
Add Terms Of Use. Closes #225

### LINKED ISSUE #225: Terms of use modal
Hypergraph terms of use (see https://www.notion.so/Terms-2794527482b54eddb5b7c991e8152999) still need to be implemented in the app. The chosen implementation is to have them as the first page of the welcome modals with a button to accept.

I can do the implementation, but there is a design necessary for this screen. @chartgerink has made a proposal here: https://www.figma.com/file/kf9PzbeOIIsmCJfjtjUr8P/welcome-modals?node-id=121%3A285. @psobrakseaton could you check this and make changes if necessary?

### PR DESCRIPTION
**Pull request by juliangruber**
Whenever the welcome screens are shown _and_ there is no profile (ie after database reset or first start), the Terms Of Use display.

I have removed the analytics checkbox for now since I'm not aware of a work item to add analytics.

Closes #225.

image

### CODE REVIEW

**Comment by psobrakseaton** [comment]
> Whenever the welcome screens are shown _and_ there is no profile (ie after database reset or first start), the Terms Of Use display.
> 
> I have removed the analytics checkbox for now since I'm not aware of a work item to add analytics.
> 
> Closes #225.
> 
> image

Weird question, but do you think that even if we don't have active analytics, having the consent for them still makes sense? @chartgerink, do you want to weigh in here?

File: app/components/welcome/terms-of-use.js [hunk 1]
```diff
@@ -0,0 +1,484 @@
+import React from 'react'
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:1]:
```suggestion
import React from 'react'
import Anchor from '../anchor'
```

For external links to open in browser, not the app.

File: app/components/welcome/terms-of-use.js [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+      </ul>
+      <h1 id='b5dfbe5d-5971-41ce-982e-61a03cec2aa4' className=''>
+        Copyright
+      </h1>
+      <p id='d142db2f-520f-4a65-88d7-8db292b6ac3a' className=''>
+        You agree that content you share through Hypergraph is made publicly
+        available under the{' '}
+        <a href='https://creativecommons.org/publicdomain/zero/1.0/legalcode'>
+          CC0 Public Domain Dedication
+        </a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:220]:
```suggestion
        <Anchor href='https://creativecommons.org/publicdomain/zero/1.0/legalcode'>
          CC0 Public Domain Dedication
        </Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+      <p id='9c716f35-1e43-45e0-a041-c0bfb9e98d75' className=''>
+        You agree to accept sole responsibility for the content you share
+        through Hypergraph. You agree to not hold liable Liberate Science for
+        any consequences of your use of the Hypergraph software.
+      </p>
+      <p id='592fed41-28fb-4a38-baf3-17bee013a4d1' className=''>
+        Content provided to you under a{' '}
+        <a href='https://creativecommons.org/publicdomain/zero/1.0/legalcode'>
+          CC0 Public Domain Dedication
+        </a>{' '}
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:241]:
```suggestion
        <Anchor href='https://creativecommons.org/publicdomain/zero/1.0/legalcode'>
          CC0 Public Domain Dedication
        </Anchor>{' '}
```

File: app/components/welcome/terms-of-use.js [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+      <p id='e5f466a0-6bcd-44db-aec9-590e87a46534' className=''>
+        Formal takedown notices can be sent to{' '}
+        <a href='mailto:copyright@libscie.org'>copyright@libscie.org</a> and
+        must enumerate all claims as exactly as possible. By sending a takedown
+        notice, ou agree to public disclosure of the takedown notice (we will
+        redact any information regarding natural persons). We will not take into
+        account any claims that do not have a{' '}
+        <a href='https://en.wikipedia.org/wiki/Burden_of_proof_(law)#Preponderance_of_the_evidence'>
+          preponderance of evidence
+        </a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:263]:
```suggestion
        <Anchor href='https://en.wikipedia.org/wiki/Burden_of_proof_(law)#Preponderance_of_the_evidence'>
          preponderance of evidence
        </Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+        We make a distinction between active data collection and passive data
+        collection.{' '}
+      </p>
+      <p id='92d13c01-3ce8-45e4-82a7-042612172854' className=''>
+        We use the following third parties to provide you with functionality in
+        Hypergraph and process the resulting data:
+      </p>
+      <ul id='857438b3-a167-466d-9c39-733317c1d16b' className='bulleted-list'>
+        <li>
+          <a href='https://aws.amazon.com/privacy/'>Amazon Web Services</a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:299]:
```suggestion
          <Anchor href='https://aws.amazon.com/privacy/'>Amazon Web Services</Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+        </li>
+      </ul>
+      <ul id='18778524-d942-4f07-93b5-64375a4d1b15' className='bulleted-list'>
+        <li>
+          <a href='https://chatra.com/privacy-policy/'>Chatra</a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:304]:
```suggestion
          <Anchor href='https://chatra.com/privacy-policy/'>Chatra</Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 7]
```diff
@@ -0,0 +1,484 @@
...
+        </li>
+      </ul>
+      <ul id='8c5cea07-af49-4976-9a56-06832bac9101' className='bulleted-list'>
+        <li>
+          <a href='https://cloud68.co/legal/privacy-policy'>Cloud68</a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:309]:
```suggestion
          <Anchor href='https://cloud68.co/legal/privacy-policy'>Cloud68</Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 8]
```diff
@@ -0,0 +1,484 @@
...
+        </li>
+      </ul>
+      <ul id='47682cc6-4d2d-4e24-9a55-21c7274bd3dd' className='bulleted-list'>
+        <li>
+          <a href='https://www.digitalocean.com/legal/privacy-policy/'>
+            DigitalOcean
+          </a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:316]:
```suggestion
          <Anchor href='https://www.digitalocean.com/legal/privacy-policy/'>
            DigitalOcean
          </Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 9]
```diff
@@ -0,0 +1,484 @@
...
+        </li>
+      </ul>
+      <ul id='e6650b5f-4749-4e4e-9cc5-f1a0fcfe97da' className='bulleted-list'>
+        <li>
+          <a href='https://policies.google.com/privacy'>Google App Suite</a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:321]:
```suggestion
          <Anchor href='https://policies.google.com/privacy'>Google App Suite</Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 10]
```diff
@@ -0,0 +1,484 @@
...
+        </li>
+      </ul>
+      <ul id='d314ef68-2c3f-41a4-86ae-63c69d289a05' className='bulleted-list'>
+        <li>
+          <a href='https://www.notion.so/Privacy-Policy-3468d120cf614d4c9014c09f6adc9091'>
+            Notion
+          </a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:328]:
```suggestion
          <a href='https://www.notion.so/Privacy-Policy-3468d120cf614d4c9014c09f6adc9091'>
            Notion
          </a>
```
```suggestion
          <Anchor href='https://www.notion.so/Privacy-Policy-3468d120cf614d4c9014c09f6adc9091'>
            Notion
          </Anchor>
```

File: app/components/welcome/terms-of-use.js [hunk 11]
```diff
@@ -0,0 +1,484 @@
...
+        </li>
+      </ul>
+      <ul id='10ac46a8-1b9c-44f1-9217-7e6605ac612c' className='bulleted-list'>
+        <li>
+          <a href='https://stripe.com/en-de/privacy'>Stripe</a>
```
Comment (chartgerink) [inline app/components/welcome/terms-of-use.js:333]:
```suggestion
          <Anchor href='https://stripe.com/en-de/privacy'>Stripe</Anchor>
```

**Review by chartgerink — CHANGES_REQUESTED** [review]
The links didn't open in the browser. I think the suggestions fix that?

Usage statistics/analytics was part of another work item that I own, which was why I mentioned it in the issue for the Terms. 

Could you please add the checkbox? The copy is:

> Allow statistics to be sent to Liberate Science GmbH

Default would be checked, and if it is checked upon "agree" [this linked code in Notion](https://www.notion.so/libscie/Implement-website-desktop-analytics-a1736fde6d174ddb93068c8a54d80951#a2a1b1ce9a1d4ade82e30cbe35b68d0f) should be injected into the head.

**Comment by juliangruber** [comment]
> The links didn't open in the browser. I think the suggestions fix that?

I forgot about those, good catch!

I implemented it differently however, since your solution would make it hard to update the Terms if they ever change, since the Anchors need to be manually replaced every time.

**Comment by juliangruber** [comment]
> Could you please add the checkbox? The copy is:
> 
> > Allow statistics to be sent to Liberate Science GmbH
> 
> Default would be checked, and if it is checked upon "agree" [this linked code in Notion](https://www.notion.so/libscie/Implement-website-desktop-analytics-a1736fde6d174ddb93068c8a54d80951#a2a1b1ce9a1d4ade82e30cbe35b68d0f) should be injected into the head.

This is a separate work item, I'm going to open an issue for this. If we add the checkbox now but don't add the code behind it yet, there's a broken feature in the app

**Comment by chartgerink** [comment]
> > The links didn't open in the browser. I think the suggestions fix that?
> 
> I forgot about those, good catch!
> 
> I implemented it differently however, since your solution would make it hard to update the Terms if they ever change, since the Anchors need to be manually replaced every time.

Yes, that's great! Thanks. 

I tried reviewing again; I don't see any newer commits since last time?

**Comment by juliangruber** [comment]
> I tried reviewing again; I don't see any newer commits since last time?

Yeah, when someone requests a review that can also mean they ask you to reconsider your last review, after new information has been shared. Before I re-requested it, your state was still "requests changes".

**Comment by chartgerink** [comment]
Ah okay :) I agree with the plan of action. 

The links still open in the electron environment, so I guess I'll not approve just yet?

**Comment by juliangruber** [comment]
Dang I'm sorry, forgot to push

**Review by chartgerink — APPROVED** [review]

**Review by psobrakseaton — APPROVED** [review]

**Pull request merged**